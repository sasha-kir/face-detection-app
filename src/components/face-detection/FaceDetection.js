import React from 'react';
import './FaceDetection.css';

const FaceDetection = ({ boundingBoxes, imageToDisplay }) => {
	let renderedBoxes = [];

	if (boundingBoxes.length > 0) {
		renderedBoxes = boundingBoxes.map((box, index) =>
			<div key={index}
				 className="bounding-box" 
				 style={{top: box.topRow, bottom: box.bottomRow, 
						 right: box.rightCol, left: box.leftCol}}>
			</div>
		);
	};

	if (imageToDisplay) {
		return (
			<div className="center-content">
				<div className="relative mt3 mb1">
					<img id="input-image" src={imageToDisplay} height="400px" width="auto" alt="" />
					{renderedBoxes}
				</div>
			</div>
		);
	} else {
		return null;
	}
}

export default FaceDetection;