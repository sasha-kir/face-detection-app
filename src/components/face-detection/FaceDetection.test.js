import React from 'react';
import { mount, shallow } from 'enzyme';
import FaceDetection from './FaceDetection';

describe('FaceDetection', () => {
    it('renders correctly', () => {
        const component = shallow(<FaceDetection boundingBoxes={[]} imageToDisplay="" />);
        expect(component).toMatchSnapshot();
    });
    it('does not display image when url is empty', () => {
        const component = shallow(<FaceDetection boundingBoxes={[]} imageToDisplay="" />);
        expect(component.exists('#input-image')).toBeFalsy();
    });
    it('displays image when url is provided', () => {
        const url = 'https://picsum.photos/500/';
        const component = shallow(<FaceDetection boundingBoxes={[]} imageToDisplay={url} />);
        const source = component.find('#input-image').prop('src');
        expect(source).toBe(url);
    });
    it('renders one bounding box', () => {
        const boxes = [ { leftCol: 0, rightCol: 0, topRow: 0, bottomRow: 0 } ];
        const url = 'https://picsum.photos/500/';
        const component = shallow(<FaceDetection boundingBoxes={boxes} imageToDisplay={url} />);
        expect(component.find('.bounding-box')).toHaveLength(1);
    });
    it('renders multiple bounding boxes', () => {
        const boxes = [ { leftCol: 0, rightCol: 0, topRow: 0, bottomRow: 0 },
                        { leftCol: 0.5, rightCol: 0.5, topRow: 0.5, bottomRow: 0.5 } 
                      ];
        const url = 'https://picsum.photos/500/';
        const component = shallow(<FaceDetection boundingBoxes={boxes} imageToDisplay={url} />);
        expect(component.find('.bounding-box')).toHaveLength(2);
    });
});