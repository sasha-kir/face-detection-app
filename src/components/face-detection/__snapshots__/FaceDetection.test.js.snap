// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaceDetection renders correctly 1`] = `""`;
